src/herbpy/action/rogue.py

```diff
@@ -0,0 +1,321 @@
+                 This must be the right arm.
+    @param render Render tsr samples during planning
+    """
+    #Presenting an object
+    if type(focus) == openravepy.openravepy_int.KinBody:
+        focus_trans = focus.GetTransform()
+    #Presenting a point in space
+    elif (type(focus) == numpy.ndarray) and (focus.ndim == 1):
+        if len(focus) != 3:
+            raise prpy.exceptions.PrPyExceptions('A presentation in space \
+                    must contain exactly three coordinates')
+        focus_trans = numpy.eye(4)
+        focus_trans[0:3, 3] = focus
+    else:
+        raise prpy.exceptions.PrPyException('Focus of the presentation is an \
+                unknown object')
+
+    if manip is None:
+        print "Setting the right arm to be the active manipulator."
+        manip = robot.right_arm
+        manip.SetActive()
+
+    if manip.GetName() != 'right':
+        raise prpy.exceptions.PrPyException('Presenting is only defined \
+                on the right arm.')
+
+    present_tsr = robot.tsrlibrary(None, 'present', focus_trans, manip)
+    
+    with prpy.viz.RenderTSRList(present_tsr, robot.GetEnv(), render=render):
+        robot.PlanToTSR(present_tsr, execute=True)
+    
+    robot.right_hand.MoveHand(f1=1, f2=1, f3=1, spread=3.14)
+
+@ActionMethod
+def Sweep(robot, start, end, manip=None, margin=0.3, render=True):
+    """
+    @param robot The robot performing the sweep
+    @param start The object or 3-d position that marks the start
+    @param end The object of 3-d position that marks the end
+    @param manip The manipulator to perform the sweep
+    @param margin The distance between the start object and the hand,
+                  so the vertical space between the hand and objects. 
+                  This must be enough to clear the objects themselves.
+    @param render Render tsr samples during planning
+    """
+
+    if type(start) == openravepy.openravepy_int.KinBody: 
+        start_trans = start.GetTransform()
+        start_coords = start_trans[0:3, 3]
+    elif (type(start) == numpy.ndarray) and (start.ndim == 1):
+        if len(start) != 3:
+            raise prpy.exceptions.PrPyExceptions('A point in space must \
+                    contain exactly three coordinates')
+        start_coords = start
+    else: 
+        raise prpy.exceptions.PrPyException('Focus of the point is an \
+                unknown object')
+
+    if type(end) == openravepy.openravepy_int.KinBody:
+        end_trans = end.GetTransform()
+        end_coords = end_trans[0:3, 3]
+    elif (type(end) == numpy.ndarray) and (end.ndim == 1):
+        if len(end) != 3:
+            raise prpy.exceptions.PrPyExceptions('A point in space must \
+                    contain exactly three coordinates')
+        end_coords = end
+        end_trans = numpy.eye(4)
+        end_trans[0:3, 3] = end_coords
+    else:
+        raise prpy.exceptions.PrPyException('Focus of the point is an \
+                unknown object')
```

Comment:
Same as above. In fact, there is a lot of code duplication here. I wonder if you could factor out this logic into a helper function that contains the switch-case statement for input types?